### ISSUE TITLE
local LND connection fails e.g. umbrel.local:8080

### ISSUE THREAD

**Issue by pseudozach** [getAlby/lightning-browser-extension#252]
Not sure if there's way to disable cert checks in an extension or install these certificates but currently I'm unable to connect local LND.

> Failed to load resource: net::ERR_CERT_AUTHORITY_INVALID

**Comment by bumi** [comment]
yes, this is sadly still an issue and we need to write guides for this. 
Browsers are sadly very strict with self-signed certificates and it is not possible to ignore the check (which is bad anyway) and also not possible to add the cert for the requests. 

For this we think we currently have the following options: 

1. configure letsencrypt certs (LND has support for automated Let's Encrypt certificates)
2. use a proxy on the server that either listens to HTTP (bad, but maybe on the local network ok?) or that has a domain with certificate (for example caddy can be used which also does automatic letsencrypt certs)
3. use a local native companion app (which can do requests with the correct LND cert) - this is in development and will require an additional app.

Do you have any thoughts what's the easiest and best UX to solve this?

**Comment by bumi** [comment]
sadly I do not have an umbrel right now. doesn't umbrel use HTTP only on local networks?

**Comment by pseudozach** [comment]
thanks for the background. That's what I thought. I suppose there's a 4th solution. If some tor solution can be added to the extension (proxy or some other way) it would resolve this issue as well.

>doesn't umbrel use HTTP only on local networks?
yes, lnd rest would be running on umbrel.local:8080 or 192.16.0.x:8080

**Comment by bumi** [comment]
Yep, tor will solve that problem, too. (or if you use something like ip2tor - what raspiblitz also offers)

> yes, lnd rest would be running on umbrel.local:8080 or 192.16.0.x:8080

if that's on HTTP, then you should be able to use http://umbrel.local:8080 ?

**Comment by pseudozach** [comment]
![Screenshot from 2021-10-05 09-25-12](https://user-images.githubusercontent.com/3731591/136063802-1226b74a-7cbf-418c-a5d6-fe7351c8dc47.png)

I'll try to troubleshoot.

**Comment by bumi** [comment]
thanks for debugging. 
yep, ok then LND expects https. and yes I don't think changing the lnd config is the way to go for node systems like umbrel. 

did Alby give you an alert when you tried to add that connection? (error handling still needs to be improved :/)

**Comment by pseudozach** [comment]
ok got it working, probably better to make this into a guide at some point:
- visit https://umbrel.local:8080 with your browser. click the lock icon on address bar and export the certificate to your computer.
- go to `chrome://settings/certificates` -> Authorities tab. 
- Click import and select your lnd REST certificate you exported earlier.
- Export your lnd admin.macaroon in hex format, there may be easier ways but this is what I did
```
ssh umbrel@umbrel.local
docker exec -it lnd /bin/sh
xxd -p -c1000 /data/.lnd/data/chain/bitcoin/mainnet/admin.macaroon
```
- Add LND details to your alby config by using `https://umbrel.local:8080` as address and hex macaroon from previous step.

**Comment by mpalladi** [comment]
I found this workaround simpler / more user friendly to get working.

1. Browse to the https://umbrel.local:8080, and accept the certificate warning - this will install it.
2. Umbrel->Connect Wallet, select Zeus and copy the Hex macaroon from there

Is it possible to use an onion address in the URL ? The main use case I have is to be able to make payments away from home when connected to umbrel.local

I have tried a Bluewallet connected to umbrel.local, but for some reason Alby won't connect with this method
